### ISSUE TITLE
RecordExps not handled by expression compiler

### ISSUE THREAD

**Issue by MattWindsor91** [UoY-RoboStar/robochart-csp-gen#115]
If I have a type set

```
record FireTarget {
	stat : FireStatus // True if this message actually describes a fire.
	// ..omitted
}
enumeration FireStatus {
	NoFire   // We did not see a fire here.
	Detected // We think there's a fire, but we're too far away to box it.
	Tracked  // We were able to box the fire and locate the centre of it.  (We might not yet be at the centre; see VNavStatus).
}
```

and a variable

`var tgt: FireTarget = FireTarget (| stat = FireStatus::NoFire |)`

this validates but crashes the CSP generator with:

```
An internal error occurred during: "Generating CSP".
Case not treated by expression compiler: class circus.robocalc.robochart.impl.RecordExpImpl
```

Given the way records lower into the CSP type system, would the `RecordExp` case be similar to a `TupleExp`?  If so, I could have a stab at making a PRQ on this later.  (I presume there's a straightforward mapping from fields to positions in the tuple.)

**Comment by pefribeiro** [comment]
Hi @MattWindsor91,

> ```
> An internal error occurred during: "Generating CSP".
> Case not treated by expression compiler: class circus.robocalc.robochart.impl.RecordExpImpl
> ```

Yes, I think this is not implemented at the moment. I don't recall whether there were any concerns with implementing this, but perhaps @alvarohm can comment.
 
> Given the way records lower into the CSP type system, would the `RecordExp` case be similar to a `TupleExp`? If so, I could have a stab at making a PRQ on this later. (I presume there's a straightforward mapping from fields to positions in the tuple.)

I think it is similar, except that record types get mapped to datatypes in CSPM. There may be some additional work to be done, eg. ensuring that used variables in the expression of a constructor are accounted for properly. This would need some tests.

**Comment by pefribeiro** [comment]
@MattWindsor91, I've implemented this in commit fd5a081 given I had already touched similar code yesterday and this seemed easy enough. When @alvarohm merges https://github.com/UoY-RoboStar/robochart-csp-gen/pull/114 this will become available.
